src/MySqlEventStore.php

```diff
@@ -393,151 +353,71 @@ public function delete(StreamName $streamName): void
+        $statement->setFetchMode(PDO::FETCH_OBJ);
+        $statement->execute($values);
 
-        return $this->defaultProjectionFactory;
-    }
+        if ($statement->errorCode() !== '00000') {
+            $errorCode = $statement->errorCode();
+            $errorInfo = $statement->errorInfo()[2];
 
-    public function getDefaultReadModelProjectionFactory(): ReadModelProjectionFactory
-    {
-        if (null === $this->defaultReadModelProjectionFactory) {
-            $this->defaultReadModelProjectionFactory = new PdoEventStoreReadModelProjectionFactory(
-                $this->connection,
-                $this->eventStreamsTable,
-                $this->projectionsTable
+            throw new RuntimeException(
+                "Error $errorCode. Maybe the event streams table is not setup?\nError-Info: $errorInfo"
             );
         }
 
-        return $this->defaultReadModelProjectionFactory;
-    }
+        $result = $statement->fetchAll();
 
-    public function deleteProjection(string $name, bool $deleteEmittedEvents): void
-    {
-        $sql = <<<EOT
-UPDATE $this->projectionsTable SET status = ? WHERE name = ? LIMIT 1;
-EOT;
+        $streamNames = [];
 
-        if ($deleteEmittedEvents) {
-            $status = ProjectionStatus::DELETING_INCL_EMITTED_EVENTS()->getValue();
-        } else {
-            $status = ProjectionStatus::DELETING()->getValue();
+        foreach ($result as $streamName) {
+            $streamNames[] = new StreamName($streamName->real_stream_name);
         }
 
-        $statement = $this->connection->prepare($sql);
-        $statement->execute([
-            $status,
-            $name,
-        ]);
-    }
-
-    public function resetProjection(string $name): void
-    {
-        $sql = <<<EOT
-UPDATE $this->projectionsTable SET status = ? WHERE name = ? LIMIT 1;
-EOT;
-
-        $statement = $this->connection->prepare($sql);
-        $statement->execute([
-            ProjectionStatus::RESETTING()->getValue(),
-            $name,
-        ]);
-    }
-
-    public function stopProjection(string $name): void
-    {
-        $sql = <<<EOT
-UPDATE $this->projectionsTable SET status = ? WHERE name = ? LIMIT 1;
-EOT;
-
-        $statement = $this->connection->prepare($sql);
-        $statement->execute([
-            ProjectionStatus::STOPPING()->getValue(),
-            $name,
-        ]);
+        return $streamNames;
     }
 
-    public function fetchStreamNames(
-        ?string $filter,
-        bool $regex,
+    public function fetchStreamNamesRegex(
+        string $filter,
         ?MetadataMatcher $metadataMatcher,
-        int $limit,
-        int $offset
+        int $limit = 20,
+        int $offset = 0
     ): array {
-        if (null === $filter && $regex) {
-            throw new Exception\InvalidArgumentException('No regex pattern given');
-        }
-
-        if ($regex && false === @preg_match("/$filter/", '')) {
+        if (false === @preg_match("/$filter/", '')) {
```

Comment:
This would also fail with an empty $filter string.
`preg_match("//", '');` is fine, but
`SELECT 1 REGEXP '' ERROR 1139 (42000): Got error 'empty (sub)expression' from regexp`